Automattic/wp-calypso
# InlineHelp: getInlineHelpSearchResultsForQuery() should not receive query as a second argument

fixes https://github.com/Automattic/wp-calypso/issues/43549

#### Changes proposed in this Pull Request

- `getInlineHelpSearchResultsForQuery()` pulls the current search from the
state instead of taking an optional second argument of which search to
look for.  Tests updated to not pass second argument.

- Also updated the docs to reflect that the function can never return
null, in bad cases it returns an empty array instead (always).

#### Testing instructions 

* CLI:  `yarn run test-client -- client/state/inline-help/test/selectors.js`

* UI: The inline help search box should continue to display results after a successful search.

![domain-box](https://user-images.githubusercontent.com/937354/86048408-186c0280-ba16-11ea-8a09-056507f91bf1.png)

If the selector doesn't work, no results are displayed, even if the search was successful.  If you see results, you are using the selector, so you know it works.
